tests/cloner_test.go

```diff
@@ -521,6 +521,107 @@ var _ = Describe("Block PV Cloner Test", func() {
 	})
 })
 
+var _ = Describe("Namespace with quota", func() {
+	f := framework.NewFrameworkOrDie(namespacePrefix)
+	var (
+		orgConfig *v1.ResourceRequirements
+		sourcePvc *v1.PersistentVolumeClaim
+		targetPvc *v1.PersistentVolumeClaim
+	)
+
+	BeforeEach(func() {
+		By("Capturing original CDIConfig state")
+		config, err := f.CdiClient.CdiV1alpha1().CDIConfigs().Get(common.ConfigName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		orgConfig = config.Spec.PodResourceRequirements
+	})
+
+	AfterEach(func() {
+		By("Restoring CDIConfig to original state")
+		config, err := f.CdiClient.CdiV1alpha1().CDIConfigs().Get(common.ConfigName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		config.Spec.PodResourceRequirements = orgConfig
+		_, err = f.CdiClient.CdiV1alpha1().CDIConfigs().Update(config)
+		Expect(err).ToNot(HaveOccurred())
+		if sourcePvc != nil {
+			By("[AfterEach] Clean up source PVC")
+			err := f.DeletePVC(sourcePvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+		if targetPvc != nil {
+			By("[AfterEach] Clean up target PVC")
+			err := f.DeletePVC(targetPvc)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	It("Should create clone in namespace with quota", func() {
+		f.CreateQuotaInNs(int64(1), int64(1024*1024*1024), int64(2), int64(2*1024*1024*1024))
+		smartApplicable := f.IsSnapshotStorageClassAvailable()
+		sc, err := f.K8sClient.StorageV1().StorageClasses().Get(f.SnapshotSCName, metav1.GetOptions{})
+		if err == nil {
+			value, ok := sc.Annotations["storageclass.kubernetes.io/is-default-class"]
+			if smartApplicable && ok && strings.Compare(value, "true") == 0 {
+				Skip("Cannot test host assisted cloning for within namespace when all pvcs are smart clone capable.")
```

Comment:
Smart/dumb cloning should be irrelevant to quota, right?

Reply:
No because smart cloning doesn't spawn pods. So we can only test host assisted cloning.

Reply:
But the new PVC may cause a quota violation, right?  Is that tested?

Reply:
Not on the memory or cpu limits that we are talking about here. There might be storage quotas but those are out of scope for this.